src/plugins/objects/restobject.ts

```diff
@@ -0,0 +1,184 @@
+import type RestChannel from 'common/lib/client/restchannel';
+import type * as Utils from 'common/lib/util/utils';
+import type {
+  GetObjectParams,
+  PrimitiveOrObjectReference,
+  RestCompactObjectData,
+  RestLiveObject,
+  RestObjectOperation,
+  RestObjectPublishResult,
+} from '../../../ably';
+
+export class RestObject {
+  channel: RestChannel;
+
+  constructor(channel: RestChannel) {
+    this.channel = channel;
+  }
+
+  /**
+   * Read object data. Defaults to { compact: true } and entrypoint=root when no id is provided.
+   * Returns undefined if provided objectId/path does not resolve to an object.
+   */
+  async get(params?: Omit<GetObjectParams, 'compact'> & { compact?: true }): Promise<RestCompactObjectData | undefined>;
+  async get(params: Omit<GetObjectParams, 'compact'> & { compact: false }): Promise<RestLiveObject | undefined>;
+  async get(params?: GetObjectParams): Promise<RestCompactObjectData | RestLiveObject | undefined> {
+    const client = this.channel.client;
+    const format = client.options.useBinaryProtocol ? client.Utils.Format.msgpack : client.Utils.Format.json;
```

Comment:
We don't currently support msgpack format REST responses on the realtime side, so I think we should always use JSON here for now